packages/fluentui/react-northstar/src/components/Pill/Pill.tsx

```diff
@@ -0,0 +1,162 @@
+import * as customPropTypes from '@fluentui/react-proptypes';
+import * as _ from 'lodash';
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+
+import {
+  UIComponentProps,
+  ContentComponentProps,
+  commonPropTypes,
+  createShorthandFactory,
+  SizeValue,
+  createShorthand,
+} from '../../utils';
+import { ComponentEventHandler, ShorthandValue, FluentComponentStaticProps } from '../../types';
+import { BoxProps } from '../Box/Box';
+
+import {
+  ComponentWithAs,
+  useAccessibility,
+  getElementType,
+  useStyles,
+  useTelemetry,
+  useFluentContext,
+  useUnhandledProps,
+} from '@fluentui/react-bindings';
+import { PillContent } from './PillContent';
+
+export interface PillProps extends UIComponentProps, ContentComponentProps<ShorthandValue<BoxProps>> {
+  /**
+   * A Pill can be sized.
+   */
+  size?: Extract<SizeValue, 'smaller' | 'small' | 'medium'>;
+
+  /**
+   * A Pill can be rounded
+   */
+  rounded?: boolean;
+
+  /**
+   * A Pill can be filled, inverted or outline
+   */
+  appearance?: 'filled' | 'inverted' | 'outline';
+
+  /**
+   * Media slot
+   */
+  media?: ShorthandValue<BoxProps>;
+
+  /**
+   * Details that will appear as second line
+   */
+  details?: ShorthandValue<BoxProps>;
+
+  /** A Pill can be disabled. */
+  disabled?: boolean;
+
+  /** A Pill can be selected. */
+  selected?: boolean;
+
+  /** A callback to be called when Pill is clicked */
+  onClick?: ComponentEventHandler<PillProps>;
```

Comment:
```suggestion
```

I don't see any special handling for `onClick`, can we remove this until we will need it?